src/reconcile/rec_write.c

```diff
@@ -555,6 +560,14 @@ __rec_write_status(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
 		return (EBUSY);
 
 	/*
+	 * If there is only one block and no update was used, no progress has
+	 * been made and there is no point swapping the new page into place.
+	 */
+	if (r->bnd_next == 1 && F_ISSET(r, WT_EVICT_UPDATE_RESTORE) &&
+	    r->bnd[0].supd != NULL && !r->update_used)
+		return (EBUSY);
+
+	/*
```

Comment:
The issue for checkpoints is more wide-ranging: imagine we call `__rec_write_status`, then `__rec_write_wrapup` fails: the checkpoint will fail but we've lost our chance to set the tree's `modified` flag back to true. Or maybe I'm just missing something...